s1tbx-op-sar-processing/src/main/java/org/esa/s1tbx/sar/gpf/BandPassFilterOp.java

```diff
@@ -0,0 +1,430 @@
+        category = "Radar/SAR Utilities",
+        authors = "Esteban Aguilera, Carlos Hernandez",
+        version = "1.0",
+        copyright = "Copyright (C) 2021 by SENSAR",
+        description = "Creates a basebanded SLC based on a subband of 1/3 the original bandwidth")
+public class BandPassFilterOp extends Operator {
+
+    @SourceProduct
+    private Product sourceProduct;
+
+    @TargetProduct
+    private Product targetProduct;
+
+    @Parameter(valueSet = {"low", "high"},
+            defaultValue = "low",
+            label = "Subband")
+    private String subband = "low";
+
+    @Parameter(valueSet = {"0.5", "0.75", "0.8", "0.9", "1"},
+            description = "Hamming alpha",
+            defaultValue = "1",
+            label = "Hamming alpha")
+    private double alpha = 1;
+
+    // Source and target bands
+    private final Map<Band, Band> targetSourceMap = new HashMap<>(10);
+    private final Map<Band, Band> complexSrcMap = new HashMap<>(10);
+    private final Map<Band, Band> complexTgtMap = new HashMap<>(10);
+
+    // Values to be loaded on initialization
+    private double sourceAlpha;
+    private double rangeSamplingRate;
+    private double subbandCenterFrequency;
+    private double subbandBandwidth;
+
+    // Constants
+    private static final String PRODUCT_SUFFIX_LOW = "_low_subband";
+    private static final String PRODUCT_SUFFIX_HIGH = "_high_subband";
+
+    /**
+     * Default constructor. The graph processing framework
+     * requires that an operator has a default constructor.
+     */
+    public BandPassFilterOp() {
+
+    }
+
+    /**
+     * Initializes this operator and sets the one and only target product.
+     * <p>The target product can be either defined by a field of type {@link Product} annotated with the
+     * {@link TargetProduct TargetProduct} annotation or
+     * by calling {@link #setTargetProduct} method.</p>
+     * <p>The framework calls this method after it has created this operator.
+     * Any client code that must be performed before computation of tile data
+     * should be placed here.</p>
+     *
+     * @throws OperatorException If an error occurs during operator initialisation.
+     * @see #getTargetProduct()
+     */
+    @Override
+    public void initialize() throws OperatorException {
+
+        try {
+            // Load metadata
+            getProductMetadata();
+
+            // Create target product
+            createTargetProduct();
+        } catch (Throwable e) {
+            OperatorUtils.catchOperatorException(getId(), e);
+        }
+    }
+
+    private double getSourceAlpha(MetadataElement abs) {
+
+        final String windowType = abs.getAttributeString("range_window_type");
+        if (windowType != null && windowType.equals("Hamming")) {
+            return abs.getAttributeDouble("range_window_coefficient");
+        } else {
+            SystemUtils.LOG.warning("The source product is assumed to have no spectral windowing applied.");
```

Comment:
For all missions other than Sentinel1 this condition would apply?

Reply:
For now, this feature will be used with Sentinel-1 and ALOS only. If we require support for other missions in the future, we would need to update the respective product reader, and if required, support other windows in the BandPassFilterOp.